### PR TITLE
bugfix: should initialize ContainerIO for all container

### PR DESCRIPTION
**Pull request by fuweid**
### Ⅰ. Describe what this PR did

For now, when pouchd restarts, the restore function only re-connects to
existing running/pause containers with initializing container IO.
However, the stopped container also needs the container IO if the user
want to restart the container. For the case, we need to initialize
container IO for all existing container, not just for running/pause
containers.

Signed-off-by: Wei Fu <fuweid89@gmail.com>

### Ⅱ. Does this pull request fix one issue?

none

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

add one

### Ⅳ. Describe how to verify it

CI

### CODE REVIEW

File: test/z_cli_daemon_test.go [hunk 1]
```diff
@@ -470,6 +470,47 @@ func (suite *PouchDaemonSuite) TestDaemonWithMultiRuntimes(c *check.C) {
 	dcfg2.KillDaemon()
 }
 
+// TestRestartStoppedContainerAfterDaemonRestart is used to test the case that
+// when container is stopped and then pouchd restarts, the restore logic should
+// initialize the existing container IO settings even though they are not alive.
+func (suite *PouchDaemonSuite) TestRestartStoppedContainerAfterDaemonRestart(c *check.C) {
+	cfg := daemon.NewConfig()
```
Comment (rudyfly) [inline test/z_cli_daemon_test.go]: Is start daemon with config file? or with command line as before?
And we need think about alios test.

File: daemon/mgr/container.go [hunk 2]
```diff
@@ -256,19 +256,20 @@ func (mgr *ContainerManager) Restore(ctx context.Context) error {
...
 		return errors.Wrap(err, "failed to get container list")
 	}
 
 	// start recover all alive containers
 	for _, c := range containers {
-		logrus.Debugf("Start recover container %s", c.Key())
 		id := c.Key()
-		// recover the running or paused container.
+
+		// NOTE: when pouch restarts, we need to initialize container IO
```
Comment (rudyfly) [inline daemon/mgr/container.go]: s/restarts/is restarting

File: daemon/mgr/container.go [hunk 3]
```diff
@@ -280,6 +281,13 @@ func (mgr *ContainerManager) Restore(ctx context.Context) error {
 			return err
 		}
 
+		// recover the running or paused container.
+		if !(c.IsRunning() || c.IsPaused()) {
```
Comment (rudyfly) [inline daemon/mgr/container.go]: running or paused containers don't to recover?

File: daemon/mgr/container.go [hunk 4]
```diff
@@ -256,19 +256,20 @@ func (mgr *ContainerManager) Restore(ctx context.Context) error {
...
-			}})
+		},
+	)
 	if err != nil {
-		logrus.Errorf("failed to get container list when restore alive containers: %v", err)
+		logrus.Errorf("failed to get container list when restore containers: %v", err)
 		return errors.Wrap(err, "failed to get container list")
 	}
 
 	// start recover all alive containers
```
Comment (HusterWan) [inline daemon/mgr/container.go]: delete this line

File: daemon/mgr/container.go (hunk 3 above)
Comment (fuweid) [reply daemon/mgr/container.go]: no need to recover the created/stopped container

File: daemon/mgr/container.go (hunk 4 above)
Comment (fuweid) [reply daemon/mgr/container.go]: will do.

File: test/z_cli_daemon_test.go (hunk 1 above)
Comment (fuweid) [reply test/z_cli_daemon_test.go]: synced with @rudyfly offline. @rudyfly is right. we would use specific config file when we write daemon related case. If not, the `/etc/pouch/config.json` will impact the running case. 

updated it.

**Comment by rudyfly** [comment]
LGTM

**Pull request merged**